daml-lf/spec/transaction.rst

```diff
@@ -358,6 +370,16 @@ identifier.
 
 ``key_with_maintainers`` is optional. 
 
+(*since version 12*)
+
+As of version 12, these fields are included:
+
+* `message Identifier`_ template_id
+* `message VersionedValue`_ arg_unversioned
+* ``string`` agreement
+
+``template_id`` and ``arg_unversioned`` is required while
```

Comment:
This seems to be a bit confusing, you state above that the contract instance is required and then here you say it’s optional for 12. I’d put the two statements close to each other to make it clear that it’s required in 10 and the alternative is arg_unversioned in 12.

Reply:
This is the original design of the document. I do not feel like changing it now. 
Furthermore it is stated at the beginning: 

```
Do not read this without ``transaction.proto``
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

In the proto definition, there is `since` and `until` comments that make that clear.